Description: Add kubuntu-text theme
Author: Jonathan Riddell <jriddell@ubuntu.com>
Forwarded: not-needed
Last-Update: 2011-01-21

Index: b/configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -296,6 +296,7 @@
            themes/script/Makefile
            themes/ubuntu-logo/Makefile
            themes/ubuntu-text/Makefile
+           themes/kubuntu-text/Makefile
            images/Makefile
 	   scripts/plymouth-generate-initrd
 	   scripts/plymouth-populate-initrd
Index: b/themes/Makefile.am
===================================================================
--- a/themes/Makefile.am
+++ b/themes/Makefile.am
@@ -1,2 +1,2 @@
-SUBDIRS = spinfinity fade-in text details solar glow script ubuntu-logo ubuntu-text
+SUBDIRS = spinfinity fade-in text details solar glow script ubuntu-logo ubuntu-text kubuntu-text
 MAINTAINERCLEANFILES = Makefile.in
Index: b/themes/kubuntu-text/kubuntu-text.plymouth
===================================================================
--- /dev/null
+++ b/themes/kubuntu-text/kubuntu-text.plymouth
@@ -0,0 +1,11 @@
+[Plymouth Theme]
+Name=Kubuntu Text
+Description=Text mode theme based on kubuntu-logo theme
+ModuleName=ubuntu-text
+
+[ubuntu-text]
+title=Kubuntu 11.04
+black=0x0078C2
+white=0xffffff
+brown=0x009DFD
+blue=0x988592
Index: b/themes/kubuntu-text/Makefile.am
===================================================================
--- /dev/null
+++ b/themes/kubuntu-text/Makefile.am
@@ -0,0 +1,4 @@
+themedir = $(datadir)/plymouth/themes/kubuntu-text
+dist_theme_DATA = kubuntu-text.plymouth
+
+MAINTAINERCLEANFILES = Makefile.in
