Description: Add kubuntu-text theme
Author: Jonathan Riddell <jriddell@ubuntu.com>
Forwarded: not-needed
Last-Update: 2012-04-12

Index: trunk/configure.ac
===================================================================
--- trunk.orig/configure.ac
+++ trunk/configure.ac
@@ -446,6 +446,7 @@
            themes/script/Makefile
            themes/ubuntu-logo/Makefile
            themes/ubuntu-text/Makefile
+           themes/kubuntu-text/Makefile
            images/Makefile
 	   scripts/plymouth-generate-initrd
 	   scripts/plymouth-populate-initrd
Index: trunk/themes/Makefile.am
===================================================================
--- trunk.orig/themes/Makefile.am
+++ trunk/themes/Makefile.am
@@ -1,2 +1,2 @@
-SUBDIRS = spinfinity fade-in text details solar glow script spinner ubuntu-logo ubuntu-text
+SUBDIRS = spinfinity fade-in text details solar glow script spinner ubuntu-logo ubuntu-text kubuntu-text
 MAINTAINERCLEANFILES = Makefile.in
Index: trunk/themes/kubuntu-text/kubuntu-text.plymouth.in
===================================================================
--- /dev/null
+++ trunk/themes/kubuntu-text/kubuntu-text.plymouth.in
@@ -0,0 +1,11 @@
+[Plymouth Theme]
+Name=Kubuntu Text
+Description=Text mode theme based on kubuntu-logo theme
+ModuleName=ubuntu-text
+
+[ubuntu-text]
+title=Kubuntu 12.04
+black=0xB2BAC1
+white=0x000000
+brown=0xA4BAC0
+blue=0x988592
Index: trunk/themes/kubuntu-text/Makefile.am
===================================================================
--- /dev/null
+++ trunk/themes/kubuntu-text/Makefile.am
@@ -0,0 +1,4 @@
+themedir = $(datadir)/plymouth/themes/kubuntu-text
+dist_theme_DATA = kubuntu-text.plymouth.in
+
+MAINTAINERCLEANFILES = Makefile.in
